File: src/main/java/itmo/is/project/mapper/trade/TradePolicyMapper.java
package itmo.is.project.mapper.trade;

import itmo.is.project.dto.trade.policy.TradePolicyDto;
import itmo.is.project.dto.trade.policy.UpdateTradePolicyRequest;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.trade.TradePolicy;
import org.mapstruct.Mapper;

@Mapper
public interface TradePolicyMapper extends EntityMapper<TradePolicyDto, TradePolicy> {
    TradePolicy toEntity(UpdateTradePolicyRequest request);
}

File: src/main/kotlin/itmo/isproject/trade/mapper/TradePolicyMapper.kt
package itmo.isproject.trade.mapper

import itmo.isproject.trade.dto.policy.TradePolicyDto
import itmo.isproject.trade.dto.policy.UpdateTradePolicyRequest
import itmo.isproject.shared.common.mapper.EntityMapper
import itmo.isproject.trade.model.TradePolicy
import org.mapstruct.Mapper

@Mapper
interface TradePolicyMapper : EntityMapper<TradePolicyDto, TradePolicy> {

    fun toEntity(request: UpdateTradePolicyRequest): TradePolicy
}
